deployment/deployment.sh

```diff
@@ -0,0 +1,225 @@
+                  if [[ $choice == [Nn] ]]; then
+                    sudo subscription-manager repos --enable codeready-builder-for-rhel-9-${arch}-rpms
+                  elif [[ $choice == [Yy] ]]; then
+                    sudo dnf install dnf-plugins-core
+                    sudo dnf config-manager --set-enabled codeready-builder-for-rhel-9-*-rpms
+                  fi
+                  curl -O https://cdn.rstudio.com/r/rhel-9/pkgs/R-${R_VERSION}-1-1.x86_64.rpm
+                  sudo dnf install -y R-${R_VERSION}-1-1.x86_64.rpm
+                  ;;
+                8)
+                  sudo yum install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+                  ask_question "Is this Linux instance running in a Public Cloud? (AWS, Azure, GCP, etc)"
+                  # Check if the choice is 'n', if so, exit the script
+                  if [[ $choice == [Nn] ]]; then
+                    sudo subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-rpms
+                  elif [[ $choice == [Yy] ]]; then
+                    sudo dnf install dnf-plugins-core
+                    sudo dnf config-manager --set-enabled "codeready-builder-for-rhel-8-*-rpms"
+                  fi
+                  curl -O https://cdn.rstudio.com/r/centos-8/pkgs/R-${R_VERSION}-1-1.x86_64.rpm
+                  sudo yum install -y R-${R_VERSION}-1-1.x86_64.rpm
+                  ;;
+            esac
+            ;;
+        Ubuntu)
+            sudo apt-get update
+            sudo apt-get install gdebi-core
+            case $ubuntu_version in
+                  22)
+                  curl -O https://cdn.rstudio.com/r/ubuntu-2204/pkgs/r-${R_VERSION}_1_amd64.deb
+                  sudo gdebi -n r-${R_VERSION}_1_amd64.deb
+                  ;;
+                  20)
+                  curl -O https://cdn.rstudio.com/r/ubuntu-2004/pkgs/r-${R_VERSION}_1_amd64.deb
+                  sudo gdebi -n r-${R_VERSION}_1_amd64.deb
+                  ;;
+            esac
+
+    esac
+fi
+
+echo "Verify R Installation"
+R --version
+
+echo "Adding R/Rscript to path"
+export PATH=/opt/R/${R_VERSION}/bin/:$PATH
+
+cd fsbench/ || exit
+
+export TARGET_DIR=$directory
+
+if [ -z "$OUTPUT_FILE" ]; then
+    export OUTPUT_FILE="/opt"
+fi
+
+ask_question "Do you want to run the setup for fsbench?"
+
+# Check if the choice is 'n', if so, exit the script
+if [[ $choice == [Nn] ]]; then
+    echo "Exiting script."
+    exit 1
+fi
+
+make setup
+
+  # Prompt the user for yes/no input
+ask_question "Start fsbench run for storage mounted at ${directory}?"
+
+# Check if the choice is 'n', if so, exit the script
+if [[ $choice == [Nn] ]]; then
+    echo "Exiting script."
+    exit 1
+fi
+
+echo "Outputting log to directory ${OUTPUT_FILE}"
+
+make
+
+echo "fsbench has completed it's run on ${OUTPUT_FILE}"
+
+  # Prompt the user for yes/no input
+ask_question "Do you want to remove R ${R_VERSION}?"
```

Comment:
It's going to ask this even if R was already installed?

Reply:
I ask every time because if the script has errored prior to this iteration after the step of installing R,  I  want to provide the option to remove R.

Reply:
I think it's going to feel extremely weird for someone who has R installed already. I'd feel super weird about it if someone sent me an R script that offered to uninstall R.

Is it necessary for this all to be one script? What if you added installing/uninstalling R as makefile options?

---

## Prerequisites

fsbench requires R to be installed on the local machine. If you don't have it installed, you can easily install it with:

```
make install-r
```

You can specify a specific R version to install/uninstall by setting the `R_VERSION` environment variable, for example, `R_VERSION=4.3.2 make install-r`.

To uninstall R,

```
make uninstall-r
```

Reply:
It's not necessary for it to be all one script, but it does add a convenience factor for SE/Support if they ask a customer admin to test their storage so that it doesn't accidentally result in leftover artifacts on the system. That's why I haven't updated the README to include any references to this deployment script, it seems like something that should be run with caution, especially if we ask folks to run it on a live system, which we often do. 

Maybe a good option would be to create a temporary file in the repo directory if the user has chosen to install R via this deployment script, then only ask about the uninstall if that file exists? Basically as a double check that this R installation is from the script?